dask/utils.py

```diff
@@ -757,6 +748,15 @@ def dispatch(self, cls):
                 register()
                 self._lazy.pop(toplevel, None)
                 return self.dispatch(cls)  # recurse
+            try:
+                impl = lk[cls2]
+            except KeyError:
+                pass
+            else:
+                if cls is not cls2:
+                    # Cache lookup
+                    lk[cls] = impl
+                return impl
```

Comment:
The lazy registration is sometimes registering more specific types which the dispatch misses otherwise. I should probably add a dedicated test for this / break it into a dedicated PR.

This shows up when running some of the tests in isolation